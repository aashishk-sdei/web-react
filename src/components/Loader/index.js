import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";
import SpinnerImg from "../../assets/images/spinner.gif";
const styles = () => ({
  default: {
    color: "#FAFAFA",
  },
  primary: {
    color: "#295DA1",
  },
  secondary: {
    color: "#212122",
  },
  danger: {
    color: "#FC4040",
  },
  indeterminate: {
    animationDuration: "550ms",
  },
});
const Loader = ({ classes, color, spinner, ...props }) => {
  const getLoaderColor = () => {
    switch (color) {
      case "default":
        return classes.default;
      case "secondary":
        return classes.secondary;
      case "danger":
        return classes.danger;
      case "primary":
      default:
        return classes.primary;
    }
  };
  const loaderColor = getLoaderColor();
  return (
    <div className="h-full flex justify-center items-center">
      {spinner ? (
        <CircularProgress
          color="primary"
          disableShrink
          classes={{
            root: loaderColor,
            indeterminate: classes.indeterminate,
          }}
          {...props}
        />
      ) : (
        <img src={SpinnerImg} alt="storied" className="w-10 h-10" />
      )}
    </div>
  );
};
Loader.propTypes = {
  color: PropTypes.oneOf(["default", "primary", "secondary", "danger"]),
  spinner: PropTypes.bool,
};
Loader.defaultProps = {
  color: "primary",
  spinner: false,
};
export default withStyles(styles)(Loader);
