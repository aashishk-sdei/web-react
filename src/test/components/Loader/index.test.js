import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../../../components/Loader';


const color = "primary";
test('Test Loader component', () => {
    const component = shallow(<Loader/>);
    const loaderExists = component.find('Loader').exists();
    expect(loaderExists).toBe(true)
});

describe("Loader testing", () => {
    it("Loader component renders properly", () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper).toMatchSnapshot();
      });

    it('check loader default color props', () => {
        const wrapper = shallow(<Loader color={color}/>);
        expect(wrapper.props().color).toEqual(color);
    })

})
  