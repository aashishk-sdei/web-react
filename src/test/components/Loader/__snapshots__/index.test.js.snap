// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader testing Loader component renders properly 1`] = `ShallowWrapper {}`;
